client/shipping-label/state/reducer.js

```diff
@@ -328,25 +329,39 @@ reducers[ OPEN_ADD_ITEM ] = ( state ) => {
 	return {
 		...state,
 		showAddItemDialog: true,
+		addedItems: {},
 	};
 };
 
 reducers[ CLOSE_ADD_ITEM ] = ( state ) => {
 	return {
 		...state,
-		movedItemIndex: -1,
 		showAddItemDialog: false,
 	};
 };
 
-reducers[ SET_ADDED_ITEM ] = ( state, { sourcePackageId, movedItemIndex } ) => {
+reducers[ SET_ADDED_ITEM ] = ( state, { sourcePackageId, movedItemIndex, added } ) => {
 	return {
 		...state,
-		sourcePackageId,
-		movedItemIndex,
+		addedItems: { ...state.addedItems,
+			[ sourcePackageId ]: added
+				? { ...state.addedItems[ sourcePackageId ], [ movedItemIndex ]: true }
+				: _.omit( state.addedItems[ sourcePackageId ], movedItemIndex )
```

Comment:
I don't think the code is much cleaner (though I'm open to suggestions), but I made the change for the sake of having the state be a bit cleaner.